src/hyp3_sdk/stac/stac.py

```diff
@@ -0,0 +1,573 @@
+        }
+    )
+
+    item = _create_item(base_url, mid_time, geo_info, insar_products, extra_properties)
+    thumbnail = base_url.replace('.zip', '_unw_phase.png')
+    item.add_asset(
+        key='thumbnail',
+        asset=pystac.Asset(href=thumbnail, media_type=pystac.MediaType.PNG, roles=['thumbnail']),
+    )
+    item.stac_extensions += [sat.SatExtension.get_schema_uri(), view.ViewExtension.get_schema_uri()]
+    item.validate()
+    return item
+
+
+def _create_rtc_stac_item(job: Job, geo_info: GeoInfo, available_polarizations: Iterable[str]) -> pystac.Item:
+    """Create a STAC item from a HyP3 RTC product.
+
+    Args:
+        job: A HyP3 Job object
+        geo_info: A GeoInfo object containing the geospatial information of the product
+        available_polarizations: A list of the available polarizations
+
+    Returns:
+        A STAC item for the product
+    """
+    base_url = job.to_dict()['files'][0]['url']
+    pattern = '%Y%m%dT%H%M%S'
+    date_string = base_url.split('/')[-1].split('_')[2].split('.')[0]
+    start_time = datetime.strptime(date_string, pattern).replace(tzinfo=timezone.utc)
+    lineage = 'Created by ASF DAAC HyP3 using the hyp3_gamma plugin running GAMMA SAR processing software'
+    extra_properties = {
+        'sar:product_type': job.to_dict()['job_type'],
+        'sar:polarizations': available_polarizations,
+        'processing:level': 'L3',
+        'processing:lineage': lineage,
+    }
+    item = _create_item(base_url, start_time, geo_info, available_polarizations + RTC_PRODUCTS, extra_properties)
+    thumbnail = base_url.replace('.zip', '_rgb_thumb.png')
+    item.add_asset(
+        key='thumbnail',
+        asset=pystac.Asset(href=thumbnail, media_type=pystac.MediaType.PNG, roles=['thumbnail']),
+    )
+    item.validate()
+    return item
+
+
+def _create_item(
+    base_url: str, start_time: datetime, geo_info: GeoInfo, product_types: Iterable[str], extra_properties: dict
+) -> pystac.Item:
+    """Create a STAC item from a HyP3 product.
+
+    Args:
+        base_url: The base url for the product
+        start_time: The start time of the product
+        geo_info: A GeoInfo object containing the geospatial information of the product
+        product_types: A list of the product types available
+        extra_properties: A dictionary of extra properties to add to the STAC item
+
+    Returns:
+        A STAC item for the product
+    """
+    properties = {
+        'data_type': 'float32',
+        'nodata': 0,
+        'proj:shape': geo_info.shape,
+        'proj:transform': geo_info.proj_transform,
+        'proj:epsg': geo_info.epsg,
+        'sar:instrument_mode': 'IW',
+        'sar:frequency_band': sar.FrequencyBand.C,
+        'sar:observation_direction': 'right',
+    }
+    properties.update(extra_properties)
+    item = pystac.Item(
+        id=base_url.split('/')[-1].replace('.zip', ''),
+        geometry=geo_info.bbox_geojson,
```

Comment:
Preferably `geometry` is the valid data footprint rather than the bbox. could bring in this dependency for convenience https://stactools.readthedocs.io/en/stable/footprint.html